crates/core/src/db/queries/metadatas.rs

```diff
@@ -85,87 +157,176 @@ pub fn list(
+        );
     }
 
-    if let Some(owners) = owners {
-        query = query.filter(current_metadata_owners::owner_address.eq(any(owners)));
+    if let Some(offerers) = offerers {
+        let mut bid_receipts_query = Query::select()
+            .columns(vec![
+                (BidReceipts::Table, BidReceipts::Metadata),
+                (BidReceipts::Table, BidReceipts::Price),
+            ])
+            .from(BidReceipts::Table)
+            .and_where(Expr::col((BidReceipts::Table, BidReceipts::Buyer)).is_in(offerers))
+            .and_where(Expr::tbl(BidReceipts::Table, BidReceipts::PurchaseReceipt).is_null())
+            .and_where(Expr::tbl(BidReceipts::Table, BidReceipts::CanceledAt).is_null())
+            .take();
+
+        if let Some(auction_houses) = auction_houses {
+            bid_receipts_query.and_where(
+                Expr::col((BidReceipts::Table, BidReceipts::AuctionHouse)).is_in(auction_houses),
+            );
+        }
+
+        query.join_lateral(
+            JoinType::InnerJoin,
+            bid_receipts_query.take(),
+            BidReceipts::Table,
+            Expr::tbl(BidReceipts::Table, BidReceipts::Metadata)
+                .equals(Metadatas::Table, Metadatas::Address),
+        );
     }
 
-    if let Some(offerers) = offerers {
-        query = query
-            .filter(bid_receipts::buyer.eq(any(offerers)))
-            .filter(bid_receipts::purchase_receipt.is_null())
-            .filter(bid_receipts::canceled_at.is_null());
+    if let Some(attributes) = attributes {
+        for AttributeFilter { trait_type, values } in attributes {
+            let alias = format!("attributes_{}", trait_type);
+            let alias: DynIden = SeaRc::new(Alias::new(&alias));
+
+            query.join_lateral(
+                JoinType::InnerJoin,
+                Query::select()
+                    .from(Attributes::Table)
+                    .column((Attributes::Table, Attributes::MetadataAddress))
+                    .and_where(Expr::col(Attributes::TraitType).eq(trait_type))
+                    .and_where(Expr::col(Attributes::Value).is_in(values))
+                    .take(),
+                alias.clone(),
+                Expr::tbl(alias, Attributes::MetadataAddress)
+                    .equals(Metadatas::Table, Metadatas::Address),
+            );
+        }
     }
 
-    if listed {
-        query = query.filter(not(listing_receipts::price.is_null()));
+    if let Some(collection) = collection {
+        query.inner_join(
+            MetadataCollectionKeys::Table,
+            Expr::tbl(
+                MetadataCollectionKeys::Table,
+                MetadataCollectionKeys::MetadataAddress,
+            )
+            .equals(Metadatas::Table, Metadatas::Address),
+        );
+
+        query.and_where(
+            Expr::col((
+                MetadataCollectionKeys::Table,
+                MetadataCollectionKeys::CollectionAddress,
+            ))
+            .eq(collection),
+        );
```

Comment:
@imabdulbasit does this look correct? The caller passes in collection address ie the mint address of the NFT and then inner join the collection address into the the metadata and filter out any that don't match the provided collection address.

Reply:
@dandlezzz @imabdulbasit and I would like to use the mint address of the collection nft as the filter parameter. He mentioned you wanted nft address. Can you share why nft address over mint address?

Reply:
We can use either as filter, shouldn't be a problem but collection mint_address is a field in token metadata.